### ISSUE TITLE
Failure on valid syntax in feature files

### ISSUE THREAD

**Issue by cranphin** [cucumber/cucumber-eclipse#278]
This is a valid string for cucumber:
"I enter the initial password {string} and confirmation {string}"

But with cucumber-eclipse-plugin-0.0.22-SNAPSHOT.zip it fails:
```
java.util.regex.PatternSyntaxException: Illegal repetition near index 28
I enter the initial password {string} and confirmation {string}
                            ^
	at java.util.regex.Pattern.error(Unknown Source)
	at java.util.regex.Pattern.closure(Unknown Source)
	at java.util.regex.Pattern.sequence(Unknown Source)
	at java.util.regex.Pattern.expr(Unknown Source)
	at java.util.regex.Pattern.compile(Unknown Source)
	at java.util.regex.Pattern.<init>(Unknown Source)
	at java.util.regex.Pattern.compile(Unknown Source)
	at cucumber.eclipse.steps.integration.Step.setText(Step.java:27)
	at cucumber.eclipse.steps.jdt.StepDefinitions.getCukeSteps(StepDefinitions.java:225)
	at cucumber.eclipse.editor.steps.jdt.JDTStepDefinitions.collectCukeStepsFromSource(JDTStepDefinitions.java:123)
	at cucumber.eclipse.editor.steps.jdt.JDTStepDefinitions.getSteps(JDTStepDefinitions.java:84)
	at cucumber.eclipse.editor.steps.ExtensionRegistryStepProvider.reloadSteps(ExtensionRegistryStepProvider.java:45)
	at cucumber.eclipse.editor.steps.ExtensionRegistryStepProvider.<init>(ExtensionRegistryStepProvider.java:28)
	at cucumber.eclipse.editor.editors.Editor.doSetInput(Editor.java:177)
```

I guess it tries to parse it as Java regexp, then fails, and doesn't also try it as Cucumber expression.
See also: https://docs.cucumber.io/cucumber/cucumber-expressions/

**Comment by qvdk** [comment]
Duplicate of #267

**Comment by qvdk** [comment]
Fixed with PR #285
